src/Symfony/Bundle/SecurityBundle/Templating/Helper/LogoutUrlHelper.php

```diff
@@ -97,6 +111,14 @@ public function getLogoutUrl($key)
      */
     private function generateLogoutUrl($key, $referenceType)
     {
+        // Fetch the current provider key from token, if possible
+        if (null === $key && $this->tokenStorage) {
+            $token = $this->tokenStorage->getToken();
+            if ($token && method_exists($token, 'getProviderKey')) {
+                $key = $token->getProviderKey();
+            }
+        }
+
         if (!array_key_exists($key, $this->listeners)) {
             throw new \InvalidArgumentException(sprintf('No LogoutListener found for firewall key "%s".', $key));
```

Comment:
@fabpot the feature ia about handling the current context. By specifying null, it will use the current provider key so we can reuse a twig template with `logout_url` and/or not dependable anymore of the firewall name. If a default key cannot be found, the invalidargumentexception on the lines below will be triggered as a fallback anyway.